pt/orm/saving-data.rst

```diff
@@ -1,28 +1,1228 @@
+
+    // No controller
+    $articles = TableRegistry::get('Articles');
+
+    // Pular validação na associação de Tags e
+    // Definino 'signup' como método de validação para Comments.Users
+    $entity = $articles->newEntity($this->request->getData(), [
+        'associated' => [
+            'Tags' => ['validate' => false],
+            'Comments.Users' => ['validate' => 'signup']
+        ]
+    ]);
+
+O capitulo :ref:`using-different-validators-per-association` possui mais informações
+sobre como usar diferentes validadores para associações ao transformar em entidades.
+
+O diagrama a seguir fornece uma visão geral do que acontece dentro dos métodos 
+``newEntity()`` ou ``patchEntity()``:
 
 .. figure:: /_static/img/validation-cycle.png
    :align: left
    :alt: Flow diagram showing the marshalling/validation process.
 
-.. figure:: /_static/img/save-cycle.png
-   :align: left
-   :alt: Flow diagram showing the save process.
+Você sempre pode contar de obter uma entidade de volta com ``newEntity()``. Se a validação
+falhar, sua entidade conterá erros, e quaisquer campos inválidos não serão preenchidos 
+na entidade criada.
+
+Convertendo Dados de Associação BelongsToMany
+-----------------------------
+
+Se você está salvando associações belongsToMany, você pode tanto usar uma lista de entidades
+ou uma lista de ids. Ao usar uma lista de dados de entidade, seus dados de requisição
+devem parecer com::
+
+    $data = [
+        'title' => 'My title',
+        'body' => 'The text',
+        'user_id' => 1,
+        'tags' => [
+            ['tag' => 'CakePHP'],
+            ['tag' => 'Internet'],
+        ]
+    ];
+
+O exemplo acima criará 2 novas tags. Se você deseja associar um artigo com tags existentes,
+você pode usar uma lista de ids. Seus dados de requisição devem parecer com::
+
+    $data = [
+        'title' => 'My title',
+        'body' => 'The text',
+        'user_id' => 1,
+        'tags' => [
+            '_ids' => [1, 2, 3, 4]
+        ]
+    ];
+
+Se você precisa associar a alguns belongsToMany registros existentes, e criar novos ao
+mesmo tempo, você pode usar um formato expandido::
+
+    $data = [
+        'title' => 'My title',
+        'body' => 'The text',
+        'user_id' => 1,
+        'tags' => [
+            ['name' => 'A new tag'],
+            ['name' => 'Another new tag'],
+            ['id' => 5],
+            ['id' => 21]
+        ]
+    ];
+
+Quando os dados acima são convertidos em entidades, você terá 4 tags. As duas primeiras
+serão objetos novos, e as outras duas serão referências a registros existentes.
+
+Ao converter dados belongsToMany, você pode desativar a criação de nova entidade, usando
+a opção ``onlyIds``. Quando habilitado, esta opção restringe transformação de
+belongsToMany para apenas usar a chave ``_ids`` e ignorar todos os outros dados.
+
+.. versionadded:: 3.1.0
+    A opção ``onlyIds`` foi adicionada na versão 3.1.0
+
+Convertendo Dados de Associação HasMany
```

Comment:
@markstory, I fixed the error "ERROR: Unexpected indentation." but now I got the error: "/data/pt/orm/saving-data.rst:182: WARNING: undefined label: entities-mass-assignment (if the link has no caption the label must precede a section header)"